libntxm/common/source/sample.cpp

```diff
@@ -625,16 +625,16 @@ bool Sample::reverse(u32 startsample, u32 endsample)
 u32 Sample::getDynamicRange(void)
 {
 	if (is_16_bit == true)
-		return 32767;
+		return 0xffff;
 	else
-		return 127;
+		return 0xff;
 }
 
 u32 Sample::getMaxAmplitude(u32 startsample, u32 endsample)
 {
 	void *data = getData();
 	u32 max_smp = 0;
-	u32 dr = getDynamicRange();
+	u32 dr = getDynamicRange() / 2;
```

Comment:
that's still 32767 and 127 though. It still doesn't handle the values -128 and -32768.